### PR TITLE
add topic, sort_by and order queries to get articles endpoint

### PR DESCRIPTION
**Pull request by matthew-sheehan**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -348,4 +347,60 @@ describe("GET /api/users", () => {
...
+    test("ignores invalid queries and extracts valid ones", () => {
+        return request(app)
+            .get("/api/articles?topics=cats&order=asc")
+            .expect(200)
+            .then(({ body }) => {
+                expect(body.articles.length).toBe(13)
+                expect(body.articles).toBeSortedBy("created_at")
+            })
+    })
+    test("returns 400 bad request error when query is valid but value is not", () => {
```
Comment (lewispricey) [inline __tests__/app.test.js]: I'd say a vaid query with no values would be a 404

File: api/utils.js [hunk 2]
```diff
@@ -8,4 +8,9 @@ exports.checkCategoryExists = (table, column, value) => {
                 return Promise.reject({ status: 404, msg: "Not Found"})
             }
         })
-}
\ No newline at end of file
+}
+
+exports.getValidParams = (table, column) => {
```
Comment (lewispricey) [inline api/utils.js:13]: This is a really cool util function 👍

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work on this ticket Matt, I really like the tests you've written here and you've kept the code super clean! That util function is great 😎 

I've left a few comments on the code but I'd say you're just missing a few tests (which in your model it looks like you've accounted for)
400 - invalid sort by query - ?sort_by=bananas
400 - invalid order query - ?order=bananas
200 - valid topic that exists but has no articles  (I think an example of this is ?topic=paper)

Please could you add in these extra tests and then send though another PR :)

**Review by itskatherine — APPROVED** [review]
Nice additional tests Matt, good to merge! 💪

**Pull request merged**